anaconda_project/internal/plugins/registry.py

```diff
@@ -0,0 +1,204 @@
+# -*- coding: utf-8 -*-
+# ----------------------------------------------------------------------------
+# Copyright © 2016, Continuum Analytics, Inc. All rights reserved.
+#
+# The full license is in the file LICENSE.txt, distributed with this software.
+# ----------------------------------------------------------------------------
+"""The plugin registry (used to locate plugins)."""
+from __future__ import absolute_import, print_function
+
+import os
+from os.path import (isdir, join, exists)
+
+from anaconda_project import verbose
+from .code_runner import CodeRunner
+
+
+class Plugin(object):
+    """Plugins base class."""
+
+    def __init__(self, path):
+        self.path = path
+        self.name = None
+        self._source = None
+
+        # initialize plugin preparing it to run. I.e. loading code and making
+        # sure the code is ready to be loaded
+        self.init_plugin()
+
+        # load the plugin by running the code and making sure that it
+        # implements the plugins API requirements
+        self.load_plugin()
+
+    @staticmethod
+    def is_plugin_candidate_path(path):
+        """Return True if the path is a potential plugin path."""
+        if (isdir(path) and exists(join(path, 'plugin.py'))) or path.endswith('.py'):
+            return True
+        return False
+
+    @property
+    def failed(self):
+        """Return if True if the plugin has failed to load, False otherwise.
+
+        Returns:
+
+            bool
+        """
+        return self._runner._failed
+
+    @property
+    def error(self):
+        """Return the error occurred when loading the plugin, None otherwise.
+
+        Returns:
+
+            str / None (if no error occurred)
+        """
+        return self._runner._error
+
+    @property
+    def error_detail(self):
+        """Return the error defails occurred when loading the plugin.
+        None otherwise
+
+        Returns:
+
+            str / None (if no error occurred)
+        """
+        return self._runner._error_detail
+
+    @property
+    def loggers(self):
```

Comment:
it looks like this property is only used internally by the log() method below, why have this property at all?